src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Stream.Collections.cs

```diff
@@ -382,7 +388,7 @@ public void SerializeEmptyCollection()
                 Assert.Equal("[]", JsonSerializer.Serialize(GetEmptyCollection<int>(type)));
             }
 
-            foreach (Type type in StackTypes<int>())
+            foreach (Type type in TypeHelper<int>.StackTypes)
```

Comment:
I presume this is out of scope for this PR, but here are some more asserts in loops that could be refactored. If converting these to theories would be too noisy, this might be a case where the loop can be executed to build up a collection of type/serialized-string pairs, and then have a single assert at the end that verifies the collection matches the expected output. I would aim for theories first though.

Reply:
I left these out intentionally, since this pattern is pervasive in all the `StreamTests` files. I focused my efforts on `HandleCollectionsAsync` since it's been causing us trouble both in CI and the local development loop.